cantools/database/can/templates/DBC.h

```diff
@@ -234,4 +236,78 @@ class Signal {
   uint32_t spn_ {J1939_INVALID_SPN};
 };
 
+/** Partial template specialization for std::string PhysicalDataType */
+template<typename RawDataType>
+class Signal<RawDataType, std::string> {
+ public:
+  Signal(const uint8_t* buffer, const std::string& name)
+      : buffer_(buffer),
+        name_(name) {}
+
+  Signal(const uint8_t* buffer, const std::string& name, const double offset,
+         const double scale, const std::string& data_format, const uint32_t spn)
+      : buffer_(buffer),
+        name_(name),
+        offset_(offset),
+        scale_factor_(scale),
+        data_format_(data_format),
+        spn_(spn) {}
+
+  /** Unpack signal from buffer */
+  virtual RawDataType Raw() const = 0;
+
+  /** Confirm if raw signal within acceptable range */
+  virtual bool RawInRange(const RawDataType& value) const = 0;  
+
+  /** Unpacked signal from buffer, decoded and converted to physical engineering units */
+  std::string Real() const { return Decode(Raw()); }
+
+  /** Confirm if physical engineering units value in range */
+  bool InRange(const std::string& value) const {
+    RawDataType raw = Encode(value);
+    return RawInRange(raw);
+  }
+
+  /** Decode given signal by applying scaling and offset. */
+  std::string Decode(RawDataType value) const {
+    unsigned char char_arr[sizeof(value)];
+    std::memcpy(char_arr, &value, sizeof(value));
+    std::string out(reinterpret_cast<const char*>(char_arr), sizeof(char_arr) / sizeof(char_arr[0]));
```

Comment:
I'm not sure why you need the memcpy to the `char_arr`, could you not just do the same reinterpret cast when you instantiate the string from the incoming `value`?

Reply:
Fixed